src/PartyDropper.sol

```diff
@@ -0,0 +1,120 @@
+    uint256 lastEditionId;
+
+    event EditionCreated(
+        uint256 indexed editionId,
+        address indexed creator,
+        address party,
+        string name
+    );
+    event Claimed(address indexed user, uint256 editionId);
+
+    modifier onlyAllowedCreator() {
+        require(
+            allowedCreators[msg.sender] || msg.sender == owner(),
+            "must be allowed creator or owner"
+        );
+        _;
+    }
+
+    constructor() ERC1155("") {}
+
+    // public functions
+    function mintEdition(uint256 editionId) public {
+        Edition storage fnd = editions[editionId];
+        require(
+            fnd.party.totalContributed(msg.sender) > 0,
+            "didn't contribute to PartyBid"
+        );
+        require(!editionMinter[editionId][msg.sender], "already minted");
+        editionMinter[editionId][msg.sender] = true;
+        _mint(msg.sender, editionId, 1, "");
+        emit Claimed(msg.sender, editionId);
+    }
+
+    function editionInfo(uint256 editionId)
+        public
+        view
+        returns (
+            address,
+            string memory,
+            string memory,
+            string memory
+        )
+    {
+        Edition storage fnd = editions[editionId];
+        return (address(fnd.party), fnd.name, fnd.imageURI, fnd.description);
+    }
+
+    function uri(uint256 editionId)
+        public
+        view
+        override
+        returns (string memory)
+    {
+        require(editionId <= lastEditionId, "must be created");
+        Edition storage edition = editions[editionId];
+        string memory json = Base64.encode(
+            bytes(
+                string(
+                    abi.encodePacked(
+                        '{"name": "',
+                        edition.name,
+                        '", "description": "',
+                        edition.description,
+                        '", "image": "',
+                        edition.imageURI,
+                        '"}'
+                    )
+                )
+            )
+        );
+        string memory output = string(
+            abi.encodePacked("data:application/json;base64,", json)
+        );
+        return output;
+    }
+
+    // allowedCreator functions
+    function createEdition(
+        address _party,
+        string memory _name,
+        string memory _imageURI,
+        string memory _description
+    ) public onlyAllowedCreator {
+        lastEditionId += 1;
+        IPartyBid party = IPartyBid(_party);
+        editions[lastEditionId] = Edition({
+            party: party,
+            name: _name,
+            imageURI: _imageURI,
+            description: _description
+        });
+        editionCreator[lastEditionId] = msg.sender;
+        emit EditionCreated(lastEditionId, msg.sender, address(_party), _name);
+    }
+
+    // owner functions
+    function setAllowedCreator(address _address, bool _value) public onlyOwner {
+        allowedCreators[_address] = _value;
```

Comment:
Does this require us to allowlist people manually?  Could we use the party-leader we're adding to be whitelisted here?  I guess that won't be ready in time so maybe for v2.

Reply:
I'm imaging that someone like FWB or Cryptocookout would deploy their own PartyDropper contract and then manually add creators who are allowed to create editions